dependencies/syn/src/expr.rs

```diff
@@ -1507,6 +1511,27 @@ pub(crate) mod parsing {
                     has_token,
                     rhs: Box::new(rhs),
                 });
+            } else if Precedence::HasIsMatches >= base && input.peek(Token![matches]) {
+                let matches_token: Token![matches] = input.parse()?;
+                let pat = input.parse()?;
+                let implies_token = input.parse()?;
+                let mut rhs = unary_expr(input, allow_struct)?;
+                loop {
+                    let next = peek_precedence(input);
+                    if next >= Precedence::Assign {
```

Comment:
Should this say `>= Precedence::Imply`?  Right, now:

```
enum E { A, B }
...
    assert(E::A matches E::B ==> true <==> false);
```

gets parsed like:

```
    assert(E::A matches E::B ==> (true <==> false));
```

whereas I'd expect it to get parsed like:

```
    assert((E::A matches E::B ==> true) <==> false);
```

Reply:
A similar issue, but with a different cause, is that:

```
    assert(false && E::A matches E::A ==> true);
```

gets parsed like:

```
    assert(false && (E::A matches E::A ==> true));
```

whereas I'd expect it to get parsed like:

```
    assert((false && E::A matches E::A) ==> true);
```

Reply:
Those are good points. Thank you. (The latter will take a bit longer to fix, as you might expect.)